packages/web3-eth/src/index.ts

```diff
@@ -0,0 +1,374 @@
+// Disabling because returnTypes must be last param to match 1.x params
+/* eslint-disable default-param-last */
+
+import { TransactionCall, TransactionWithSender, Web3BaseProvider } from 'web3-common';
+import { Web3Config, Web3RequestManager } from 'web3-core';
+import {
+	BlockNumberOrTag,
+	ValidTypes,
+	ValidReturnTypes,
+	convertToValidType,
+	Address,
+	Uint256,
+	HexString32Bytes,
+	HexStringBytes,
+	Uint,
+	HexString8Bytes,
+	convertObjectPropertiesToValidType,
+	isHexString32Bytes,
+	Filter,
+} from 'web3-utils';
+import {
+	convertibleBlockProperties,
+	convertibleFeeHistoryResultProperties,
+	convertibleReceiptInfoProperties,
+	convertibleTransactionInfoProperties,
+} from './convertible_properties';
+
+import * as rpcMethods from './rpc_methods';
+
+export class Web3Eth {
+	private readonly _requestManager: Web3RequestManager;
+	private readonly _options: Web3Config;
+
+	public constructor(provider: Web3BaseProvider | string, options: Web3Config) {
+		this._requestManager = new Web3RequestManager(provider);
```

Comment:
According to architecture discussed, `requestManager` instance must be used from `web3Context`.

Reply:
Yes, in the process of updating this one - just a bit confused on how to use `Web3Context` since it's an abstract class

Reply:
As discussed in architecture, this class should be inheriting web3Context through web3transaction package. we can discuss this in call today.

Reply:
Will be addressed in subsequent PR